portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,139 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions; 
+import com.google.appengine.api.datastore.FetchOptions.Builder; 
+import com.google.gson.Gson;
+import com.google.sps.data.Quote; 
+import java.util.List; 
+import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private static final String QUOTE = "Quote";
+  private static final String TEXT = "text";
+  private static final String TIMESTAMP = "timestamp"; 
+  private static final int DEFAULT_NUM_QUOTES = 5;
+
+  /**
+    * Retrives all quotes that are stored in Datastore
+    * and puts them into a list of quotes. This list is then written
+    * as the response from the server. 
+    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(QUOTE).addSort(TIMESTAMP, SortDirection.DESCENDING);
+    PreparedQuery quotes = datastore.prepare(query);
+
+    // Get the parameter for number to display, with 5 as the default value.
+    int numQuoteToDisplay = getNumQuoteDisplayed(request, DEFAULT_NUM_QUOTES);
+   
+    // Add each quote to the quote list whose content will be written as 
+    // the response from the servlet. 
+    List<Quote> quoteList = new ArrayList<>();
+    quotes.asList(FetchOptions.Builder.withLimit(numQuoteToDisplay))
+        .forEach((quoteEntity) -> {
+          quoteList.add(extractQuoteFromEntity(quoteEntity));
+        });
+    
+    // Put the thread to sleep for a short period of time.
+    // This adds buffer time for querying Datastore. 
+    try {
+      Thread.sleep(200);
```

Comment:
Do we absolutely need this? What happens if we remove the thread sleep? Please clarify these areas in the comment.

Reply:
I will add more comments regarding this sleep function.